packages/neon_framework/lib/src/storage/persistence.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart' show protected;
+
+/// A persistent key value storage.
+abstract interface class Persistence<T extends Object> {
+  /// Whether a value exists at the given [key].
+  FutureOr<bool> containsKey(String key);
+
+  /// Clears all values from persistent storage.
+  FutureOr<bool> clear();
+
+  /// Removes an entry from persistent storage.
+  FutureOr<bool> remove(String key);
+
+  /// Saves a [value] to persistent storage.
+  FutureOr<bool> setValue(String key, T value);
+
+  /// Fetches the value persisted at the given [key] from the persistent
+  /// storage.
+  FutureOr<T?> getValue(String key);
+}
+
+/// A key value persistence that caches read values to be accessed
+/// synchronously.
+///
+/// Mutating values is asynchronous.
+abstract class CachedPersistence<T extends Object> implements Persistence<T> {
+  /// Fetches the latest values from the host platform.
+  ///
+  /// Use this method to observe modifications that were made in the background
+  /// like another isolate or native code while the app is already running.
+  Future<void> reload();
+
+  /// The cache that holds all values.
+  ///
+  /// It is instantiated to the current state of the backing SQLite database
+  /// and then kept in sync via setter methods in this class.
```

Comment:
Should this really mention SQLite? It's generic enough that at some point a different database could back it.

Reply:
You're right. I thought I had already fixed this after the first review.